tests/Exercise/ValentinesDayTest.elm

```diff
@@ -0,0 +1,116 @@
+module Exercise.ValentinesDayTest exposing (tests)
+
+import Comment exposing (Comment, CommentType(..))
+import Dict
+import Exercise.ValentinesDay as ValentinesDay
+import Review.Rule exposing (Rule)
+import Review.Test
+import RuleConfig
+import Test exposing (Test, describe, test)
+import TestHelper
+
+
+tests : Test
+tests =
+    describe "ValentinesDayTest"
+        [ exemplar
+        , noCase
+        ]
+
+
+rules : List Rule
+rules =
+    ValentinesDay.ruleConfig |> .rules |> List.map RuleConfig.analyzerRuleToRule
+
+
+exemplar : Test
```

Comment:
This is a thought for the future, but the exemplar tests are always huge, and we already have the exemplar code in the elm repo. Maybe we could automatically run the analyser rules against the exemplar code for all the exercises in ci, instead of having these tests. Not sure how feasible it is, what with the code being in different repos, but feels like it could be useful ...

Reply:
It's totally feasible, we already do something like that with the test runner. I'll open an issue.

The exemplars/examples are sometimes big, that's true, but I think they have their place being duplicated in the tests. If anything, it shows what the full intended code is supposed to be and puts the failing tests in context (since we often only use the relevant snippets for those).